### ISSUE TITLE
Jasmine ReferenceError: "Map" is not defined

### ISSUE THREAD

**Issue by Geoffrey0** [searls/jasmine-maven-plugin#269]
Hello, 

I just started writing jasmine tests for a backlog of code in our project.
We frequently use the Map object as in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

However when doing a simple jasmine test it fails.

describe('MapTest', function () {
    it("tests if map is defined", function () {
        var map = new Map();
        expect(map).toBeDefined();
    });
});
Output:
  1.) MapTest it tests if map is defined <<< FAILURE!
    \* ReferenceError: "Map" is not defined.

I've tested this in version 1.3.1.3 and 1.3.1.5. 
When I use the bdd option and open the chrome browser the tests pass.
Output: 
MapTest
tests if map is defined (green)

Therefore I expect that something is wrong with our configuration. However, I can not find the issue.
Any help is greatly appreciated!

**Comment by jhwohlgemuth** [comment]
I was just passing through and I thought I would try helping out with this question:

Prior to version 2, this plugin uses [HtmlUnit](http://htmlunit.sourceforge.net/) as the browser environment in which the tests are run. Under the hood, HtmlUnit uses [Rhino](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/Rhino#Rhino_documentation) which is a *really* old ([released in 1998!](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/Rhino/History)) browser environment implemented entirely in java...as far as I know, Rhino is no longer actively supported.

In any case, [starting with version 2.0](https://searls.github.io/jasmine-maven-plugin/phantomjs.html) `jasmine-maven-plugin` uses PhantomJS by default. PhantomJS is newer than Rhino with much better JavaScript support. **However** [even PhantomJS does not implement `Map`](https://kangax.github.io/compat-table/es6/#phantom). The Rhino documentation is sparse to say the least, but I think it is safe to say it has support equal to **or less than** that of PhantomJS.

**Bottom Line**: `Map` is an [ES6 feature](https://www.ecma-international.org/ecma-262/6.0/#sec-map-objects) that is not supported by HtmlUnit or PhantomJS and therefore cannot be used for headless testing with `jasmine-maven-plugin` (via the `jasmine:test` goal). 
> **Note:** Firefox *does* support `Map` which is why your tests pass using the `jasmine:bdd` goal

This issue has irked me for a while since a few devs on my team want to use ES6, but can't because of our reliance on CI servers running tests in a headless environment... Sadly, [PhantomJS v2.5 was going to have complete ES6 support](https://github.com/ariya/phantomjs/issues/14458) but the project has recently lost its primary maintainer and I am skeptical v2.5 will ever see a "production" release...
